addons/account/models/account_move.py

```diff
@@ -872,26 +874,34 @@ def _compute_name(self):
                 continue
 
             move_has_name = move.name and move.name != '/'
-            if move_has_name or move.state != 'posted':
-                if not move.posted_before and not move._sequence_matches_date():
-                    if move._get_last_sequence():
-                        # The name does not match the date and the move is not the first in the period:
-                        # Reset to draft
-                        move.name = False
-                        continue
+            if move_has_name or move.state == 'draft':
+                if move_has_name and not move.posted_before and not move._sequence_matches_date():
```

Comment:
When do you enter here, in practice?
When do you have a name if it wasn't posted before?

Or is it to support edge cases in stable?

Reply:
On the first move of the sequence, the user is able to enter a manual name to override the sequence template. If after that he changes the date of the invoice, it's possible his override doesn't matches the dates anymore. Then we reset the name so he sees the placeholder again.